### PR TITLE
Remove StartSession component

### LINKED ISSUE #714: Move `createSession` request to Experiment component?
There is a `StartSession` component whose only responsibility is to fire a request to create a session, and pass it through to its parent (`Experiment`). @jaburgoyne Is there any scenario in which we would want to *not* create a session? It seems silly to explicitly set `StartSession` through the backend in the `first_round`, if we cannot ever proceed to the `next_round` calls, which depend on the session to be created. I would propose to get rid of that component and just fire the `createSession` call from the `Experiment` component. This avoids:
- unnecessary repetition of `StartSession` in all rules files
- the definition of a component without any html (other than error handling, which can be done at a higher level just as well)
- problems with passing through data between `Experiment` and `StartSession`

### PR DESCRIPTION
**Pull request by BeritJanssen**
Close #714 .

### CODE REVIEW

File: frontend/src/components/Experiment/Experiment.js [hunk 1]
```diff
@@ -65,12 +65,26 @@ const Experiment = ({ match }) => {
...
+            return;
+        }
+        try { 
+            createSession({
+                experiment,
+                participant,
+                playlist,
+            }).then(data => {
+                setSession(data.session);
+                setSessionStore(data.session);
```
Comment (drikusroor) [inline frontend/src/components/Experiment/Experiment.js:79]: Why do we need these both still?

File: frontend/src/components/Experiment/Experiment.js [hunk 2]
```diff
@@ -99,7 +110,7 @@ const Experiment = ({ match }) => {
         } else {
             // Try to get next_round data from server
             const round = await getNextRound({
-                session: session.current
+                session: session
```
Comment (drikusroor) [inline frontend/src/components/Experiment/Experiment.js:113]:
```suggestion
                session
```

**Review by drikusroor — APPROVED** [review]
Nice, I love it when a PR removes more code than it adds. I'm only wondering why we need to combine a local session state and a store session state. Or is this something we intend to de-duplicate later on?

File: frontend/src/components/Experiment/Experiment.js (hunk 1 above)
Comment (BeritJanssen) [reply frontend/src/components/Experiment/Experiment.js:79]: This is for "historical reasons" - I thought to leave adapting the child components of `Experiment`, some of which still expect a `session` prop, for later.

**Pull request merged**